### ISSUE TITLE
Strange graphical setting/resolution bug (and the workarounds I found)

### ISSUE THREAD

**Issue by Bruurbscat** [XLabsProject/iw4x-client#132]
When changing any graphical setting, especially noticeable with resolution, the game freezes up permanently, although for a small moment it will indeed change to the selected setting or resolution. In that freeze you see a frame of the desired changes until it freezes up. Attempting to click on it will usually not respond in any way, but sometimes clicking will result in the screen getting more and more white in Windows. Sometimes a few frames of input are detected before this happens, though this always ends up being for less than a second.
 If the game goes into a black screen instead, it will result in a DirectX crash of "undetermined error". Seems fixable through changing the desired settings through the config files found in the game folder, specifically 'players'. However, if attempting to fix it in-game, restarting the game, if it didn't go into a black screen, will show the applied changes. But anymore changes will result in the same thing. So, changing resolutions one by one and restarting the game on every change seems to be the correct solution to prevent a DirectX error, and clicking around before the first frame after the applied changes show up might cause a DirectX error anyway. If it happens while trying to do the one by one changes method, going to two by two will allow you to keep going.
Changing DirectX 9 methods has no effect. Even if not having the specific DirectX 9 in the Redist folder installed, the same thing will occur.
 In fullscreen this leads to the game being on top of everything, requiring opening the task manager through CTRL+ALT+DEL, attempting to loosely alt tabbing to the task manager and in the second that you can put an input on it, pressing the arrow keys down, and attempting to continue doing it until you can press Delete and close the .exe, which will close immediately if given enough time in this state. If the task manager was already open before this, it'll require attempting to close it and then doing the said process. This works with any game where this issue occurs, without requiring a restart.
 Here's the only minidump (.dmp) file generated. [https://cdn.discordapp.com/attachments/275267435945263104/889263330219405332/iw4x-v0.6.1-20210919163051.dmp](url)

**Comment by Dss0** [comment]
Can't reproduce this. Need more info:

- have you tried setting your game to windowed/windowed borderless mode? Does the same thing happen there?
- are you using any directx overlays or other software running in the background that hooks directx? Examples would be: msi afterburner, rivatuner statistics server, nvidia shadowplay..
- are you using any directx proxies? (as in is there a d3d9.dll or other direcx related dlls in your game folder)

Dump file shows a crash in d3d9.dll not in iw4x itself so probably hard to debug.

**Comment by Bruurbscat** [comment]
I tested Bot Warfare, launching it causes the same DirectX error. I do in fact have Afterburner and Rivatuner. Tried out the method used for Hitman Absolution of preventing issues with Afterburner and it doesn't work. Screenshot for the method from PCGW below.
 However, closing Afterburner and RivaTuner Statistics Server fixed it. Tried it with the iw4x.exe, and the two game .exe's too. Closing the issue because of that. Note, I'm using the patch for single-player and spec-ops. 
And no, I don't have any DirectX proxies.
![image](https://user-images.githubusercontent.com/91033138/134089217-057e94a7-7914-4c5c-93d6-a9bfa1be902f.png)
